File: LPAv/Trabalho 9 - Python/questao4.py
import re

stringDeEntrada = input()
while (stringDeEntrada != "####"):
    if (re.match("^[A-Za-z0-9\.\+_-]+@[A-Za-z0-9\._-]+\.[a-zA-Z]*$", stringDeEntrada)):
        print("SIM")
    else:
        print("NAO")
    stringDeEntrada = input()

File: LPAv/Trabalho  9 - C++/questao4.cpp
#include <iostream>
#include <stdio.h>
#include <string>
#include <regex>
using namespace std;

int main()
{
    string expressaoRegular;
    cin >> expressaoRegular;
    while (expressaoRegular != "####")
    {
        if (regex_match(expressaoRegular, regex("^[a-zA-Z]+@[a-zA-Z]+(\\.[a-zA-Z]+)+$")))
        {
            printf("SIM\n");
        }
        else
        {
            printf("NAO\n");
        }
        cin >> expressaoRegular;
    }
}